studio/src/studio.js

```diff
@@ -52,10 +53,24 @@ class MasStudio extends LitElement {
 
     connectedCallback() {
         super.connectedCallback();
+        this.updatePageFromUrl();
+        window.addEventListener('popstate', this.updatePageFromUrl.bind(this));
     }
 
     disconnectedCallback() {
         super.disconnectedCallback();
+        window.removeEventListener(
+            'popstate',
+            this.updatePageFromUrl.bind(this),
+        );
+    }
+
+    updatePageFromUrl() {
```

Comment:
initialization of the store from the URL should be centralized at a single place. Otherwise it will be very hard to troubleshoot.